modules/programs/wezterm.nix

```diff
@@ -0,0 +1,589 @@
+          default = { };
+          description = ''
+            The main ANSI colors.
+          '';
+          example = literalExample ''
+            {
+              black = "#000000";
+              red = "#800000";
+              green = "#008000";
+              yellow = "#808000";
+              blue = "#000080";
+              magenta = "#800080";
+              cyan = "#008080";
+              white = "#C0C0C0";
+            }
+          '';
+        };
+
+        bright = mkOption {
+          type = weztermColorschemeBase16Type false;
+          default = { };
+          description = ''
+            The main bright colors.
+          '';
+          example = literalExample ''
+            {
+              black = "#808080";
+              red = "#FF0000";
+              green = "#00FF00";
+              yellow = "#FFFF00";
+              blue = "#0000FF";
+              magenta = "#FF00FF";
+              cyan = "#00FFFF";
+              white = "#FFFFFF";
+            }
+          '';
+        };
+      };
+    };
+
+in {
+  meta.maintainers = with lib.hm.maintainers; [ l3af ];
+
+  options.programs.wezterm = with types; {
+    enable = mkEnableOption "Wezterm terminal emulator";
+
+    package = mkOption {
+      type = package;
+      default = pkgs.wezterm;
+      defaultText = "pkgs.wezterm";
+      description = ''
+        The Wezterm package to use.
+      '';
+    };
+
+    keybindings = mkOption {
+      type = listOf weztermKeybindType;
+      default = [ ];
+      description = ''
+        Mapping of keybindings to actions.
+      '';
+      example = literalExample ''
+        [{
+            modifiers = [ "SHIFT" "CTRL" ];
+            key = "l";
+            action = "wezterm.action {ActivateTabRelative = 1}";
+        }]
+      '';
+    };
+
+    mousebindings = mkOption {
+      type = listOf weztermMousebindType;
+      default = [ ];
+      description = ''
+        Mapping of mouse bindings to actions.
+      '';
+      example = literalExample ''
+        [{
+            button = "Left";
+            event = "Up";
+            count = 1;
+            modifiers = [ "CTRL" ];
+            action = "\"OpenLinkAtMouseCursor\"";
+        }]
+      '';
+    };
+
+    settings = mkOption {
+      type = attrsOf genericType;
+      default = { };
+      description = ''
+        Set any simple value.
+      '';
+      example = literalExample ''
+        {
+          enable_wayland = true;
+          font_size = 10.0;
+        }
+      '';
+    };
+
+    colors = mkOption {
+      type = weztermColorschemeType;
+      default = { };
+      description = ''
+        Set terminal colors.
+      '';
+      example = literalExample ''
+        {
+          foreground = "#C0C0C0";
+          background = "#000000";
+        }
+      '';
+    };
+
+    extraSettings = mkOption {
+      default = "";
+      type = lines;
+      description = ''
+        Additional configuration to add outside of the return statement.
```

Comment:
Adding configuration inside and outside the "return statement" seems impossible to understand unless you have intimate understanding of the implementation of the module. Perhaps it would be possible to instead assign the keybindings and such into variables and have a `script` option that the user can override, the default value would be something like
```lua
return {
  keybindings = hm_keybindings,
  …
}
```
?

Reply:
There would still be the need to be a way, unless there's a way which I can't think of, to add raw code to the actual return statement. I specify my font with `font = wezterm.font("FiraCode Nerd Font Mono"),` in `extraReturnSettings` because it calls a function which can't be passed through `settings` as strings are wrapped in quotation marks.